reconcile/gitlab_housekeeping.py

```diff
@@ -150,11 +167,12 @@ def rebase_merge_requests(dry_run, gl, rebase_limit, pipeline_timeout=None,
 
             # If pipeline_timeout is None no pipeline will be canceled
             if pipeline_timeout is not None:
-                gl_project = GitLabApi(gl_instance,
-                                       project_id=mr.source_project_id,
-                                       settings=gl_settings).project
-                clean_pipelines(dry_run, gl_project, pipelines,
-                                pipeline_timeout)
+                timeout_pipelines = \
+                    get_timeout_pipelines(pipelines, pipeline_timeout)
+                if timeout_pipelines:
```

Comment:
You don't need this `if`. The inner loop won't do anything if the list is empty.

Reply:
I want to avoid init GitLabApi if timeout_pipelines is empty.